File: Laba2-Python/Task3.py
import ConsoleInput
import ArrayHelper
import math

def Run():
    print("Завдання 3. Обчислити елементи масиву Y за формулою. Сформувати та вивести на екран новий масив R, \n" +
            "елементами якого є парні за індексом елементи вихідного (початкового) масиву Х та масиву Y.\n")

    n = int(input("Введіть кількість елементів масиву: "))

    array = generate_array(n)

    print("Массив X:")
    print_array(array)

    print("\nМассив Y:")
    print_array(get_array_y(array))

def generate_array(n):
    array = [i+1 for i in range(n)]
    return array

def print_array(array):
    for i in array:
        print(i)

def get_array_y(array_x):
    array_y = []

    for x in array_x:
        if math.cos(x) > 0:
            array_y.append(math.pow(x, 3) - 7.5)
        elif math.sin(x) <= 0:
            array_y.append(math.pow(x, 2) - (5 * math.pow(math.e, math.sin(x))))

    return array_y

run_task3()

File: Laba2/Task3.cs
using Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PZPK_Labas.Laba2
{
    public static class Task3
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 3. Обчислити елементи масиву Y за формулою. Сформувати та вивести на екран новий масив R, \n" +
                "елементами якого є парні за індексом елементи вихідного (початкового) масиву Х та масиву Y.\n");

            int n;
            ConsoleInput.SafeIntInput("Введіть кількість елементів масиву", out n);

            int[] array = ArrayHelper.GenerateArray(n);

            Console.WriteLine("Массив X:");
            ArrayHelper.PrintArray(array);

            Console.WriteLine("\nМассив Y:");
            ArrayHelper.PrintArray(GetArrayY(array));
        }

        private static double[] GetArrayY(int[] arrayX)
        {
            double[] arrayY = new double[arrayX.Length];

            for (int i = 0; i < arrayX.Length; i++)
            {
                int x = arrayX[i];

                if (Math.Cos(x) > 0)
                    arrayY[i] = Math.Round(Math.Pow(x, 3) - 7.5, 2);
                else if (Math.Sin(x) <= 0)
                    arrayY[i] = Math.Round(Math.Pow(x, 2) - (5 * Math.Pow(Math.E, Math.Sin(x))), 2);
            }

            return arrayY;
        }
    }
}
